hdfs-executor/src/main/java/org/apache/mesos/hdfs/executor/NameNodeExecutor.java

```diff
@@ -302,6 +310,31 @@ public void frameworkMessage(ExecutorDriver driver, byte[] msg) {
     log.info(String.format("Received framework message: %s", messageStr));
   }
 
+  private String getNodeId() {
+    String id = null;
```

Comment:
NodeId is based on the task type in your code, see: ```nodeId = taskName;``` So instead of using this complicated logic instead, you should just get task name from the Persistent or LiveState.